posthog/temporal/tests/batch_exports/test_s3_batch_export_workflow.py

```diff
@@ -523,8 +522,8 @@ async def s3_client(bucket_name, s3_key_prefix):
     using a disposable bucket to run these tests or sticking to other tests that use the
     local development MinIO.
     """
-    async with aioboto3.Session().client("s3") as minio_client:
-        yield minio_client
+    async with aioboto3.Session().client("s3") as s3_client:
+        yield s3_client
 
         await delete_all_from_s3(minio_client, bucket_name, key_prefix=s3_key_prefix)
```

Comment:
Is this now referencing [the module-scoped `minio_client`](https://github.com/PostHog/posthog/blob/e85168cbafd9f61ba31eefbd2343b25903255627/posthog/temporal/tests/batch_exports/test_s3_batch_export_workflow.py#L120-L138)? 🤔 I'm confused how that is working…

I tried to run these locally to poke at it myself but these tests are skipped by default so I didn't take the time to investigate further. I'm not sure if these are being run in CI or not, either -- [there are some tests that are skipped from this suite](https://github.com/PostHog/posthog/actions/runs/8266168271/job/22613553157?pr=20894), I suspect that might be these but without verbose output it's hard to say for certain.

Reply:
Yeah, this should be deleting using the `s3_client`. I missed the name update.

Tests are skipped by default when no S3 credentials are present, there is information in the README on how to run them.

Reply:
We were also not passing the S3 client down when calling `assert_clickhouse_events_in_s3`. Anyways, I've fixed both of these renaming issues, and I've also created a specific key for development testing, as my personal credentials were not working, and it's better to have a dedicated user just for testing. I've Updated the README to reflect this. S3 test is now passing.

Reply:
> Tests are skipped by default when no S3 credentials are present, there is information in the README on how to run them, as we do not run them in CI. Doing so would require populating CI with a lot of credentials. Granted, maybe that's something we will want to do in the future...

Hm, yeah… I do worry with these kinds of things that having to intentionally opt-in to running the tests makes them less likely to catch regressions, but this seems okay for now? I'd assume the MinIO targeted tests would likely catch the majority of issues anyway.

Reply:
I agree: opt-in tests means tests that are never ran. I've raised the concern already to see if we can have a better story for managing credentials in CI.